witness/src/Dscp/DB/CanProvideDB/Class.hs

```diff
@@ -1,57 +1,33 @@
 
+{-# language ExistentialQuantification #-}
+
 module Dscp.DB.CanProvideDB.Class where
 
 import Codec.Serialise
-import Control.Lens (makeLenses)
 
-type ThrowableIO m = (MonadIO m, MonadThrow m)
+type CanUseDb k v m = (Serialise k, Serialise v, MonadIO m, MonadThrow m)
+type CanDeleteKeys k m = (Serialise k, MonadIO m, MonadThrow m)
 
-data Plugin ctx = Plugin
-    { _pRun    :: forall     m a . ReaderT ctx m a -> m a
-    , _pGet    :: forall k v m   . (Serialise k, Serialise v, ThrowableIO m) =>   k      -> ReaderT ctx m (Maybe v)
-    , _pPut    :: forall k v m   . (Serialise k, Serialise v, ThrowableIO m) => [(k, v)] -> ReaderT ctx m ()
-    , _pDelete :: forall k   m   . (Serialise k,              ThrowableIO m) =>  [k]     -> ReaderT ctx m ()
+data Plugin = Plugin
+    { pGet    :: forall k v m. CanUseDb      k v m =>   k      -> m (Maybe v)
+    , pPut    :: forall k v m. CanUseDb      k v m => [(k, v)] -> m ()
+    , pDelete :: forall k   m. CanDeleteKeys k   m =>  [k]     -> m ()
```

Comment:
Is this a homegrown `Capability`?

Reply:
It is. Its the only way it can be done with SD.